File: output/gemma4-e4b/avatar/Python/Java/codeforces_408_B.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;
import java.lang.Math;

public class codeforces_408_B {

    /**
     * Counts the frequency of each character in the given string.
     * @param s The input string.
     * @return A map containing character counts.
     */
    private static Map<Character, Integer> countChars(String s) {
        Map<Character, Integer> counts = new HashMap<>();
        for (char c : s.toCharArray()) {
            counts.put(c, counts.getOrDefault(c, 0) + 1);
        }
        return counts;
    }

    /**
     * Executes the core logic: calculates the maximum number of common characters
     * between two input strings, limited by minimum frequency.
     */
    public static void solution() {
        Scanner scanner = new Scanner(System.in);
        
        // Read the first line (bought_count)
        if (!scanner.hasNextLine()) {
            return;
        }
        String line1 = scanner.nextLine();
        Map<Character, Integer> boughtCount = countChars(line1);

        // Read the second line (made_cout)
        if (!scanner.hasNextLine()) {
            // Handle case where only one line is provided, though input structure implies two.
            // We proceed assuming the second line might be empty or missing, but the logic requires it.
            // For robustness, we assume the input stream provides the second line if the first was read.
            // If the environment guarantees two lines, this check is sufficient.
        }
        String line2 = scanner.nextLine();
        Map<Character, Integer> madeCount = countChars(line2);

        int res = 0;

        // Iterate through characters present in madeCount
        for (Map.Entry<Character, Integer> entry : madeCount.entrySet()) {
            char color = entry.getKey();
            int countMade = entry.getValue();

            // Check if the color exists in boughtCount
            if (!boughtCount.containsKey(color)) {
                System.out.println(-1);
                return;
            }

            int countBought = boughtCount.get(color);
            
            // Calculate min(bought_count[color], made_cout[color])
            res += Math.min(countBought, countMade);
        }

        System.out.println(res);
        
        scanner.close();
    }

    public static void main(String[] args) {
        // The original Python code runs the solution once (t=1).
        solution();
    }

File: output/gemma4-e4b/avatar/Java/Python/codeforces_408_B.py
import sys
from collections import Counter

def is_valid(s1: str, s2: str) -> bool:
    """Checks if all characters in s2 are present in s1."""
    s1_chars = set(s1)
    for char in s2:
        if char not in s1_chars:
            return False
    return True

def solve():
    # Read all input tokens (assuming s1 and s2 are the first two tokens)
    try:
        data = sys.stdin.read().split()
        if len(data) < 2:
            # Handle case where input is insufficient
            return
        s1 = data[0]
        s2 = data[1]
    except Exception:
        return

    if not is_valid(s1, s2):
        print(-1)
        return

    # Calculate character frequencies
    m1 = Counter(s1)
    m2 = Counter(s2)

    ans = 0
    # Get unique characters from s2
    unique_chars_s2 = set(s2)

    for char in unique_chars_s2:
        # Get counts, defaulting to 0 if the character is not present
        count1 = m1.get(char, 0)
        count2 = m2.get(char, 0)
        
        # Calculate min frequency and add to answer
        min_freq = min(count1, count2)
        ans += min_freq

    print(ans)

if __name__ == "__main__":
    solve()